gadgets/trace_dns/program.bpf.c

```diff
@@ -343,6 +343,29 @@ static __always_inline int output_dns_event(struct __sk_buff *skb,
 	bpf_skb_load_bytes(skb, DNS_OFF + sizeof(struct dnshdr), event->name,
 			   name_len);
 
+	// Convert DNS string to dot notation
+	// "\u0003www\u0009wikipedia\u0003org\u0000"
+	// "www.wikipedia.org"
+	unsigned int i;
+	unsigned int remaining = 0;
+	unsigned int offset = 0;
+	for (i = 0; i < MAX_DNS_NAME - 1; i++) {
+		if (remaining == 0) {
+			remaining = event->name[i + offset];
+			offset = 1;
+			if (i > 0) {
+				if (remaining == 0) {
+					event->name[i] = '\0';
+					break;
+				}
+				event->name[i] = '.';
+				continue;
+			}
+		}
+		event->name[i] = event->name[i + 1];
+		remaining--;
+	}
+
```

Comment:
What about this:

```suggestion
	// Convert DNS string to dot notation
	// "\u0003www\u0009wikipedia\u0003org\u0000"
	// "www.wikipedia.org"
	unsigned int i;
	unsigned int remaining = event->name[0];
	for (i = 0; i < MAX_DNS_NAME - 1; i++) {
		if (remaining == 0) {
			remaining = event->name[i + 1];
			if (remaining == 0) {
				event->name[i] = '\0';
				break;
			}
			event->name[i] = '.';
			continue;
		}
		event->name[i] = event->name[i + 1];
		remaining--;
	}

```

Reply:
Thanks. I picked your patch. I also added a test if the name is empty.

I'll merge when the CI completes.